File: MedLabO.Models/TerminTest.cs
using MedLabO.Models.Test;

namespace MedLabO.Models
{
    public class TerminTest
    {
        public Guid? TerminID { get; set; }
        public Guid? TestID { get; set; }
        public TestWithoutTerminTestovi? Test { get; set; }
        public Guid? RezultatID { get; set; }
        public Rezultat? Rezultat { get; set; }
    }
}

File: MedLabO UI/medlabo_desktop/lib/models/termin_test/termin_test.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:medlabo_desktop/models/rezultat/rezultat.dart';
import 'package:medlabo_desktop/models/test/test.dart';

part 'termin_test.g.dart';

@JsonSerializable()
class TerminTest {
  String? terminID;
  String? testID;
  Test? test;
  String? rezultatID;
  Rezultat? rezultat;

  TerminTest(
    this.terminID,
    this.testID,
    this.test,
    this.rezultatID,
    this.rezultat,
  );

  factory TerminTest.fromJson(Map<String, dynamic> json) =>
      _$TerminTestFromJson(json);

  Map<String, dynamic> toJson() => _$TerminTestToJson(this);
}
